server.js

```diff
@@ -1,29 +1,126 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
+import dotenv from 'dotenv'
+import listEndpoints from 'express-list-endpoints'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const thoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: [true, 'Message required'],
+    minlength: [5, 'Min. length 5 characters'],
+    maxlength: [140, 'Max. length 140 characters']
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
+const Thought = mongoose.model('Thought', thoughtSchema)
+
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({
+      error: 'API unavailable'
+    })
+  }
+})
 
-// Start defining your routes here
+// shows list of endpoints
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// get the 20 latest thoughts, sorted from newest
+app.get('/thoughts', async (req, res) => {
+  try {
+    const thoughts = await Thought.find().sort({ createdAt: -1 }).limit(20)
+    res.json(thoughts)
+  } catch (error) {
+    res.status(400).json({ message: 'Something went wrong', error })
+  }
+})
+
+// post new thought
+app.post('/thoughts', async (req, res) => {
+  try {
+    const newThought = await new Thought(req.body).save()
+    res.json(newThought)
+  } catch (error) {
+    res.json(400).json(error)
+  }
+})
+
+// like a post
+app.post('/thoughts/:id/like', async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const likeThought = await Thought
+      .findOneAndUpdate({ _id: id }, { $inc: { hearts: 1 } }, { new: true })
```

Comment:
📄If there's many things in here it might be easier to read if they're seperated one one row each like:
{ _id: id },
{ $inc: { hearts: 1 } }
etc.